crates/wasmtime/src/runtime/vm/memory.rs

```diff
@@ -290,22 +290,6 @@ impl Memory {
         // overkill for this purpose.
         let absolute_max = 0usize.wrapping_sub(page_size);
 
-        // Sanity-check what should already be true from wasm module validation.
-        // Note that for 32-bit targets the absolute maximum is `1<<32` during
-        // compilation, not one-page-less-than-u32::MAX, so need to handle that
-        // specially here.
-        let absolute_max64 = if cfg!(target_pointer_width = "32") {
-            1 << 32
-        } else {
-            u64::try_from(absolute_max).unwrap()
-        };
-        if let Ok(size) = ty.minimum_byte_size() {
-            assert!(size <= absolute_max64);
-        }
-        if let Ok(max) = ty.maximum_byte_size() {
-            assert!(max <= absolute_max64);
-        }
```

Comment:
Why were these asserts removed?

Reply:
They were basically a pain to maintain on 32-bit because I already had to modify the minimum check to use `1<<32` rather than `absolute_max` (due to `absolute_max` being a `usize` and not being able to represent `1<<32`) and then this PR uncovered an assertion in the next one where a 64-bit linear memory's maximum size far exceeds `1<<32` as well. Given that these are just debug asserts that are already basically checked in many other places it seemed best to just remove them instead of trying to contort ourselves to keep them in all portable conditions.